src/button/createButton.ts

```diff
@@ -0,0 +1,96 @@
+import { DNode, Widget, WidgetProperties, WidgetFactory } from '@dojo/widget-core/interfaces';
+import createWidgetBase from '@dojo/widget-core/createWidgetBase';
+import { v } from '@dojo/widget-core/d';
+import themeable, { ThemeableMixin } from '@dojo/widget-core/mixins/themeable';
+import * as css from './styles/button.css';
+
+export interface ButtonProperties extends WidgetProperties {
+	/*
+	 * Text content of button
+	 */
+	content?: string;
+
+	/**
+	 * The button's name attribute
+	 */
+	name?: string;
+
+	/*
+	 * Called when the button is clicked
+	 */
+	onClick?(event: MouseEvent): void;
+
+	/*
+	 * Button type can be "submit", "reset", "button", or "menu"
+	 */
+	type?: string;
+
+	/*
+	 * Creates an icon button with the specified icon glyph
+	 */
+	icon?: string;
+
+	/*
+	 * Whether the button is disabled or clickable
+	 */
+	disabled?: boolean;
+
+	/**
+	 * ID of element with descriptive text
+	 */
+	describedBy?: string;
+
+	/**
+	 * Indicates status of a toggle button
+	 */
+	pressed?: boolean;
```

Comment:
Yup! I imagine a parent widget would need to be stateful, have `pressed: this.state['buttonPressed']` in the button properties, and use `this.setState` in an onClick handler. I'll add an example for it, as you suggested below :)